File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/PurchaseCard.cs
#region "Copyright"

//PayPal Payflow Pro .NET SDK
//Copyright (C) 2014  PayPal, Inc.
//
//This file is part of the Payflow Pro .NET SDK
//
//The Payflow .NET SDK is free software: you can redistribute it and/or modify
//it under the terms of the GNU General Public License as published by
//the Free Software Foundation, either version 3 of the License, or
//any later version.
//
//The Payflow .NET SDK is is distributed in the hope that it will be useful,
//but WITHOUT ANY WARRANTY; without even the implied warranty of
//MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//GNU General Public License for more details.
//
//You should have received a copy of the GNU General Public License
//along with the Payflow .NET SDK.  If not, see <http://www.gnu.org/licenses/>.


#endregion

#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for Purchase card related information
	/// </summary>
	/// <remarks>
	/// PurchaseCard is associated with CardTender.
	/// <seealso cref="CardTender"/>
	/// </remarks>
	public sealed class PurchaseCard : PaymentCard
	{
		#region "Member Variables"

		/// <summary>
		/// Purchase card type
		/// </summary>
		private String mCommCard;

		#endregion

		#region "Constructors"

		/// <summary>
		/// Constructor for PurchaseCard
		/// </summary>
		/// <param name="Acct">Purchase Card number</param>
		/// <param name="ExpDate">Card expiry date</param>
		/// <param name="CommCard">Purchase Card  type</param>
		/// <remarks>This is used as Payment Device for the CardTender.
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>Acct --> ACCT, ExpDate --> EXPDATE, CommCard --> COMMCARD</code>
		/// </remarks>
		/// <example>
		///  <code lang="C#" escaped="false">
		///		.............
		///		
		///		PurchaseCard PayDevice = new PurchaseCard("XXXXXXXXXX","XXXX","C");
		///		
		///		..............
		///  </code>
		///  <code lang="C#" escaped="false">
		///		.............
		///		
		///		Dim PayDevice As PurchaseCard = new PurchaseCard("XXXXXXXXXX","XXXX","C")
		///		
		///		..............
		///  </code>
		///  </example>
		///  <seealso cref="CardTender"/>
		public PurchaseCard(String Acct, String ExpDate, String CommCard) : base(Acct, ExpDate)
		{
			mCommCard = CommCard;
		}

		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			try
			{
				base.GenerateRequest();
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_COMMCARD, mCommCard));
			}
			catch (BaseException)
			{
				throw;
			}
			catch (Exception Ex)
			{
				DataObjectException DEx = new DataObjectException(Ex);
				throw DEx;
			}
            //catch
            //{
            //    throw new Exception();				
            //}
		}

		#endregion
	}

}

File: java/src/sdk/dataobjects/paypal/payflow/PurchaseCard.java
package paypal.payflow;

/**
 * Used for Purchase card related information
 * PurchaseCard is associated with CardTender. {@link CardTender}
 */
public final class PurchaseCard extends PaymentCard {

    private String commCard;

    /**
     * @param acct     Purchase Card number
     * @param expDate  Card expiry date (format mmyy)
     * @param cardType Purchase Card  type (P - Personal, C - Corprate, B - Business)
     *                 <p/>
     *  Maps to Payflow Parameter:
     * ACCT , EXPDATE , COMMCARD
     * <p/>
     * //Create the PaymentDevice object
     * PurchaseCard payDevice = new PurchaseCard("XXXXXXXXXX","XXXX","C");
     * </p>
     */
    public PurchaseCard(String acct, String expDate, String cardType) {
        super(acct, expDate);
        commCard = cardType;
    }

    /**
     * Generate the Transaction request. Overrides PaymentCard.generateRequest()
     */
    protected void generateRequest() {
        super.generateRequest();
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_COMMCARD, commCard));
    }


}
